app/src/main/java/io/bloco/template/ui/counter/CounterActivity.kt

```diff
@@ -0,0 +1,43 @@
+package io.bloco.template.ui.counter
+
+import android.os.Bundle
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.lifecycleScope
+import com.google.android.material.snackbar.Snackbar
+import io.bloco.template.R
+import io.bloco.template.common.di.ViewModelFactory
+import io.bloco.template.ui.BaseActivity
+import kotlinx.android.synthetic.main.activity_counter.*
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import javax.inject.Inject
+
+class CounterActivity : BaseActivity() {
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory<CounterViewModel>
+
+    private val viewModel by lazy {
+        ViewModelProvider(this, viewModelFactory).get(CounterViewModel::class.java)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        component.inject(this)
+        setContentView(R.layout.activity_counter)
+
+        fabIncrement.setOnClickListener { viewModel.incrementClick() }
+        fabDecrement.setOnClickListener { viewModel.decrementClick() }
+
+        viewModel.getValue()
+            .onEach {
+                value.text = it.toString()
+        }.launchIn(lifecycleScope)
```

Comment:
This indentation still feels a bit weird to me, not sure I understand what's the rule to use for streams with more than call.